src/Microsoft.Health.Dicom.Api/Features/Filters/ApiVersionValidatorAttribute.cs

```diff
@@ -0,0 +1,36 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using EnsureThat;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Dicom.Api.Features.Exceptions;
+using Microsoft.Health.Dicom.Api.Models;
+using Microsoft.Health.Dicom.Core.Configs;
+
+namespace Microsoft.Health.Dicom.Api.Features.Filters;
+
+public sealed class ApiVersionValidatorAttribute : ActionFilterAttribute
+{
+    private readonly bool _isLatestApiVersionEnabled;
+
+    public ApiVersionValidatorAttribute(
+        IOptions<FeatureConfiguration> featureConfiguration)
+    {
+        EnsureArg.IsNotNull(featureConfiguration, nameof(featureConfiguration));
+        _isLatestApiVersionEnabled = featureConfiguration.Value.EnableLatestApiVersion;
```

Comment:
Perhaps instead of simply "max," `EnableLatestApiVersion` is instead `LatestApiVersion` and it's a number

Reply:
Not exactly, if you have a "preview" txt like we had before